src/riak_kv_pb_timeseries.erl

```diff
@@ -126,6 +129,54 @@ process(#tsputreq{table = Bucket, columns = _Columns, rows = Rows}, State) ->
             {reply, missing_helper_module(Bucket, BucketProps)}
     end;
 
+%% @ignore DELETE (as if)
+process(#tsdelreq{table = Bucket, key = CompoundKey_,
+                  vclock = Vc,
+                  r = R_, w = W_, pr = PR_, pw = PW_, dw = DW_, rw = RW_,
+                  timeout = Timeout}, State) ->
+    %% we collect all the options pertinent for standard deletes
+    %% since we will follow the general delete path
+    W  = riak_pb_kv_codec:decode_quorum(W_),
+    PW = riak_pb_kv_codec:decode_quorum(PW_),
+    DW = riak_pb_kv_codec:decode_quorum(DW_),
+    R  = riak_pb_kv_codec:decode_quorum(R_),
+    PR = riak_pb_kv_codec:decode_quorum(PR_),
+    RW = riak_pb_kv_codec:decode_quorum(RW_),
+    Options =
+        lists:flatmap(
+          fun({_, undefined}) -> [];
+             ({_, default})   -> [];
+             ({K, V})         -> [{K,V}] end,
+          [{r, R}, {w, W}, {rw, RW}, {pr, PR}, {pw, PW}, {dw, DW},
+           {timeout, Timeout}]),
+
+    CompoundKey = riak_pb_ts_codec:decode_cells(CompoundKey_),
+
+    Mod = riak_ql_ddl:make_module_name(Bucket),
+    try Mod:get_ddl() of
+        DDL ->
+            PKLK = make_ts_keys(CompoundKey, DDL),
+            %% = {PK, LK} = Key to pass to riak_client:delete
+            DeleteF =
+                fun() -> riak_client:delete_vclock(
+                           Bucket, PKLK, erlify_rpbvc(Vc), Options,
+                           {riak_client, [node(), undefined]})
+                end,
+            case DeleteF() of
+                ok ->
+                    {reply, tsdelresp, State};
+                {error, notfound} ->
+                    {reply, tsdelresp, State};
+                {error, Reason} ->
+                    {reply, make_rpberrresp(
+                              ?E_DELETE, io_lib:format("Failed to delete record: ~p", [Reason])),
+                     State}
+            end
+    catch error:{undef, _} ->
+            BucketProps = riak_core_bucket:get_bucket(Bucket),
+            {reply, missing_helper_module(Bucket, BucketProps)}
```

Comment:
This should be a three-element tuple, with the service state at the end. I fixed this for other cases in riak_kv_pb_timeseries.erl this morning, so don't worry about those.